### PR TITLE
Add `id` attribute for menu

### PR DESCRIPTION
**Pull request by milwad-dev**
If you want to apply a series of animations etc. on the menu with JavaScript, this can be done much better by using ID.
Now this work is available and can be used easily.

```php
Menu::new()
    ->addId('navigation')
    ->add(Link::to('/', 'Home')->addId('home-link'))
    ->add(Link::to('/about', 'About'))
    ->add(Link::to('/contact', 'Contact'))
```

```html
<ul id="navigation">
    <li><a href="/" id="home-link">Home</a></li>
    <li><a href="/about">About</a></li>
    <li><a href="/contact">Contact</a></li>
</ul>
```

### CODE REVIEW

File: src/Html/Attributes.php [hunk 1]
```diff
@@ -8,6 +8,8 @@ class Attributes
 
     protected array $classes = [];
 
+    protected array $ids = [];
```
Comment (sebastiandedeyne) [inline src/Html/Attributes.php:11]: An element can only have one ID.
```suggestion
    protected ?string $id = null;
```

File: src/Html/Attributes.php [hunk 2]
```diff
@@ -59,10 +61,24 @@ public function addClass(string | array $class): self
         return $this;
     }
 
+    public function addId(string | array $id): self
```
Comment (sebastiandedeyne) [inline src/Html/Attributes.php:64]:
```suggestion
    public function id(?string $id): self
```

File: src/Traits/HasHtmlAttributes.php [hunk 3]
```diff
@@ -29,4 +29,11 @@ public function addClass(string $class): static
 
         return $this;
     }
+
+    public function addId(string $id): static
```
Comment (sebastiandedeyne) [inline src/Traits/HasHtmlAttributes.php:33]:
```suggestion
    public function id(?string $id): static
```

File: src/Traits/HasParentAttributes.php [hunk 4]
```diff
@@ -40,4 +40,11 @@ public function addParentClass(string $class): static
 
         return $this;
     }
+
+    public function addParentId(string $id): static
```
Comment (sebastiandedeyne) [inline src/Traits/HasParentAttributes.php:44]:
```suggestion
    public function parentId(?string $id): static
```

**Review by sebastiandedeyne — CHANGES_REQUESTED** [review]
An HTML element can only have one ID. Can the function signatures be modified to enforce this? (I posted some suggestions in the reviews)

Then we can add this in :)

**Comment by milwad-dev** [comment]
@sebastiandedeyne 
Right, I forgot this point!

**Review by sebastiandedeyne — APPROVED** [review]

**Pull request merged**